handlers.go

```diff
@@ -129,14 +128,15 @@ func (r *oauthProxy) oauthCallbackHandler(w http.ResponseWriter, req *http.Reque
 		return
 	}
 
-	client, err := r.getOAuthClient(r.getRedirectionURL(w, req), getClientAuthMethod(r.config.ClientAuthMethod))
-	if err != nil {
-		r.log.Error("unable to create a oauth2 client", zap.Error(err))
+	conf, _err := r.getOAuthConf(r.getRedirectionURL(w, req))
```

Comment:
Why `_err`? Could this not just be `err` as is the idiom?

Reply:
@JoelSpeed the reason why things like `_err` where introduced with `_` prefix, was to mark all the new changes coming from `golang/x/oauth2`, while we replace `coreos/go-oidc/oauth2`.

But thinking for a second will be less confusing to just refactor it and get rid of any references to coreos/go-oidc/oauth2.

Reply:
> But thinking for a second will be less confusing to just refactor it and get rid of any references to coreos/go-oidc/oauth2.

This is closer to what I was expecting for this PR. We should be able to validate that it works through some manual testing before we merge right?